tests/components/light/test_init.py

```diff
@@ -18,479 +18,479 @@
-from homeassistant.setup import async_setup_component, setup_component
+from homeassistant.setup import async_setup_component
 
 import tests.async_mock as mock
-from tests.common import get_test_home_assistant, mock_service, mock_storage
+from tests.common import async_mock_service
 from tests.components.light import common
 
 
-class TestLight(unittest.TestCase):
-    """Test the light module."""
+@pytest.fixture
+def mock_storage(hass, hass_storage):
+    """Clean up user light files at the end."""
+    yield
+    user_light_file = hass.config.path(light.LIGHT_PROFILES_FILE)
 
-    # pylint: disable=invalid-name
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
+    if os.path.isfile(user_light_file):
+        os.remove(user_light_file)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
 
-        user_light_file = self.hass.config.path(light.LIGHT_PROFILES_FILE)
+async def test_methods(hass):
+    """Test if methods call the services as expected."""
+    # Test is_on
+    hass.states.async_set("light.test", STATE_ON)
+    assert light.is_on(hass, "light.test")
 
-        if os.path.isfile(user_light_file):
-            os.remove(user_light_file)
+    hass.states.async_set("light.test", STATE_OFF)
+    assert not light.is_on(hass, "light.test")
 
-    def test_methods(self):
-        """Test if methods call the services as expected."""
-        # Test is_on
-        self.hass.states.set("light.test", STATE_ON)
-        assert light.is_on(self.hass, "light.test")
+    # Test turn_on
+    turn_on_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TURN_ON)
 
-        self.hass.states.set("light.test", STATE_OFF)
-        assert not light.is_on(self.hass, "light.test")
+    await common.async_turn_on(
+        hass,
+        entity_id="entity_id_val",
+        transition="transition_val",
+        brightness="brightness_val",
+        rgb_color="rgb_color_val",
+        xy_color="xy_color_val",
+        profile="profile_val",
+        color_name="color_name_val",
+        white_value="white_val",
+    )
 
-        # Test turn_on
-        turn_on_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TURN_ON)
+    await hass.async_block_till_done()
 
-        common.turn_on(
-            self.hass,
-            entity_id="entity_id_val",
-            transition="transition_val",
-            brightness="brightness_val",
-            rgb_color="rgb_color_val",
-            xy_color="xy_color_val",
-            profile="profile_val",
-            color_name="color_name_val",
-            white_value="white_val",
-        )
+    assert len(turn_on_calls) == 1
+    call = turn_on_calls[-1]
+
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_ON
+    assert call.data.get(ATTR_ENTITY_ID) == "entity_id_val"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)